File: game/tv_show/attract.py
from game_data import GameData
from phone_events import PhoneEvents
from tv_show.tv_show_resources import TvShowResources
from tv_show.video_state import VideoState


class Attract(VideoState):
    def __init__(self, context: GameData):
        super().__init__(context, TvShowResources.videos_attract[context.country], True)

    def process_events(self, phone_events: PhoneEvents):
        super().process_events(phone_events)
        from tv_show.initial import Initial
        from tv_show.playing import Playing
        from tv_show.in_cave import InCave

        if phone_events.offhook:
            return Initial

        if phone_events.press_5:
            return Playing
        if phone_events.press_6:
            self.context.forest_score = 1000
            return InCave

        return None

File: game-web/src/tvShow/attract.ts
/** Port of game/tv_show/attract.py */

import { type StateClass } from '../state';
import { TvShowResources } from './tvShowResources';
import { VideoState } from './videoState';
import type { GameData } from '../gameData';
import type { PhoneEvents } from '../phoneEvents';

// Deferred imports to avoid circular dependency issues at module load time
import { Initial } from './initial';
import { Playing } from './playing';
import { InCave } from './inCave';
import { InScoreboard } from './inScoreboard';

export class Attract extends VideoState {
  constructor(context: GameData) {
    super(
      context,
      TvShowResources.videosAttract[context.country],
      true,
      TvShowResources.audioAttract[context.country],
    );
  }

  render(ctx: OffscreenCanvasRenderingContext2D): void {
    super.render(ctx);
  }

  processEvents(events: PhoneEvents): StateClass | null {
    super.processEvents(events);

    if (events.offhook) return Initial;

    /*if (events.press_5) return Playing;
    if (events.press_4) {
      this.context.forestScore = 1000;
      this.context.forestNormalSacksCollected = 5;
      this.context.forestReachedEnd = true;
      this.context.forestGoldenSacksCollected = 1;
      this.context.forestObstaclesHit = ['1'];
      return InScoreboard;
    }
    if (events.press_6) {
      this.context.forestScore = 1000;
      return InCave;
    }*/

    return null;
  }
}
